8Knot/cache_manager/db_init.py

```diff
@@ -0,0 +1,256 @@
+"""
+NOTES ABOUT THIS FILE:
+
+This file uses raw SQL to create tables in Postgres.
+It's typically easiest and best-practice to use a db migration
+tool instead of doing error-prone manual administration like this.
+
+However, using sqlalchemy and alembic (Python db migration stack)
+would be a bit of a steep learning curve for people who just want to
+create a table in the cache. Most people who would be working on a
+project like this will know enough SQL to read the existing table
+definitions and create a new table as needed from those examples.
+
+Our data model is fairly simple, so for now the overhead of proper
+db migration tooling is mostly bloaty. We can return to this decision
+in the future if necessary.
+"""
+
+"""
+TABLE_CREATION_INSTRUCTIONS:
+
+For data that we retrieve from Postgres to be cached, we need
+to define the "schema" for the table where it will be stored.
+
+Take a moment to look at the correspondence between the table
+"commits_query" in this file and the SQL that we have the database
+execute, defined in "queries/commits_query.py." Notice that each
+of the columns referenced in the SELECT statement has a
+corresponding column in the table schema defined below, with a
+type that matches the datatype we expect to receive from Augur.
+
+Note also that we use the table creation syntax:
+CREATE UNLOGGED TABLE IF NOT EXISTS <name>
+
+We configure the tables to be unlogged to maximize write-to speed.
+We can always repopulate a table with minimal inconvenience if
+the cache database is destroyed, so we don't need the durability
+guarantees of a logged table.
+
+We also only create a new table if a table of the same name
+does not already exist. This initialization script ALWAYS runs
+on app startup to make sure that the schema of the databse is
+intact between application restarts.
+
+To add a new table to the cache, simply copy an existing table
+creation block from those below. Given that you're creating a table
+for a query in the 'queries/' folder, name the table the same name
+as the query function. Name the columns of the table, and give their
+types, and everything should work!
```

Comment:
Is there a resource that can list the type options and what they are? Not specific to 8knot, but a general resource if that makes sense?